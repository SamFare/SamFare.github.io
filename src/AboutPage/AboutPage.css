.spacer {
    height: 10vh;
}


@media only screen and (max-width: 600px) {
    .spacer {
        height: 2vh;
    }
}