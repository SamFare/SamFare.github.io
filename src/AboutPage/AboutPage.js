import React from "react";

import PhotoHeader from "../PhotoHeader/PhotoHeader.js";
import ListOfTextElement from "../ListOfTextElement/ListOfTextElement.js";
import BarChartElement from "../BarChartElement/BarChartElement.js";
import Paragraph from "../Paragraph/Paragraph.js";
import Heading from "../Heading/Heading.js";

import data from "../../data/career-data.json";



export default class AboutPage extends React.Component{
  constructor() {
    super();

    this.SkillData = [
      {skill: "ES6", skillIQ: 137},
      {skill: "c#", skillIQ: 95},
      {skill: "React", skillIQ: 200},
      {skill: "CSS", skillIQ: 150}
    ];
  }
  
  render(){
    return (
      <div>
        <PhotoHeader />
        <Heading text="About"/>
        <Paragraph />
        <Heading text="Experence"/>
        <ListOfTextElement elements={data.data} />
        <Heading text="Skills"/>
        <BarChartElement data={this.SkillData}/>
      </div>
    );
  }
}