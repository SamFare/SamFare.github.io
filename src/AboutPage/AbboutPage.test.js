import React from "react";
import AboutPage from "./AboutPage.js";
import { mount } from "enzyme";


it("The about page displays a paragraph box", () => { 
  const wrapper = mount(<AboutPage />);
  expect(wrapper.find(".qa-paragraph").length).toEqual(0);
});