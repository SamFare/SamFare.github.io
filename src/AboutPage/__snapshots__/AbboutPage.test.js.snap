// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`an about page it has the image that was passed in 1`] = `
<div>
  <div
    className="undefined qa-photo"
    style={
      Object {
        "backgroundImage": "url(assets/about-page-pic.jpg)",
      }
    }
  >
    <noscript />
    <div>
      <noscript />
    </div>
  </div>
  <div>
    <div>
      <p>
         
        About
         
      </p>
    </div>
    <div>
      I’ve been a programmer for 4 years now. Over that time, I’ve had the chance to work at some great companies including CompareTheMarket.com (current) and Philips. As a developer, I’m constantly challenging myself to learn new things. At university, my passion for learning allowed me to achieve a First, but after university, I’ve worked hard to keep learning and developing. As well as learning for myself I’m keen to feedback and share what I have learned with others. I have a blog on Medium which has CompareTheMarket.com’s most read article, and recently taught React at a ‘CodeBar’ event in London.
    </div>
    <div>
      <p>
         
        What I'm up too
         
      </p>
    </div>
    <div>
      <div>
        <div>
          <img
            height="100"
            src="assets/meta.png"
            width="100"
          />
        </div>
        <div>
          <div>
             
            Speaking at META conf
             
          </div>
          <div>
            Location: peterborough
Date: 10/05/2018

I converted my most successful medium article “I did mob programming every day for 5 months, here’s what I learnt” in to a talk at the BGL Groups all-company conference META.
          </div>
        </div>
      </div>
      <div>
        <div>
          <div>
             
            Coach at Codebar London
             
          </div>
          <div>
            Location: London
Date: 27/03/2018

CodeBar is a event aimed at helping beginner programmers develop the skills they need to suceed. During the event I acted as a coach teaching React. you can find out more information on codebar here: https://codebar.io/
          </div>
        </div>
        <div>
          <img
            height="100"
            src="assets/codebar.png"
            width="100"
          />
        </div>
      </div>
    </div>
    <div>
      <p>
         
        Experience
         
      </p>
    </div>
    <div>
      <div>
        <div>
          <img
            height="100"
            src="assets/compare-the-market-pounds.png"
            width="100"
          />
        </div>
        <div>
          <div>
             
            CompareTheMarket.com
             
          </div>
          <div>
            Key Tech: Javascript (ES6), AngularJS, C#&lt;

At comparethemarket.com I have worked hard to enhance my JavaScript skills while working on an ES6 UI for an application that supports the interaction between CTM and our panel of providers.

Comparethemarket.com has also given me the opportunity to experience new techniques such as pair and mob programming. 

I also worked on a C# 7 based project that integrates the quotes that are received from providers into the main site. This project has allowed me to deep dive into the workings of .NET including a deep look at.Net memory management for fixing a memory leak.
          </div>
        </div>
      </div>
      <div>
        <div>
          <div>
             
            General Electric
             
          </div>
          <div>
            Key Tech: Javascript (ES6), AngularJS, Magik

At GE I demonstrated my leadership and communication ability by becoming the team leader for the company’s first aiders. This involved coordinating first aiders to cover events (like the company first aid tournament) and organising the rotors to check our defibrillators and first aid kits. 

I also demonstrated my ability to pick up things quickly by working on the Smallworld product portfolio in a GE language called Magik. The Smallworld portfolio helps large utility companies look after their assets.

I also had the opportunity to work on an ES6 dashboard for reporting of our code coverage statistics. This dashboard was used by every team on site to report up.
          </div>
        </div>
        <div>
          <img
            height="100"
            src="assets/GE-logo.png"
            width="100"
          />
        </div>
      </div>
      <div>
        <div>
          <img
            height="100"
            src="assets/philips-logo.png"
            width="100"
          />
        </div>
        <div>
          <div>
             
            Philips Electronics
             
          </div>
          <div>
            Key Tech: Javascript, ruby

At philips I worked on a project monitoring ICU patients and passing the data on to doctors working remotely. This my area of this project was written in Javascript/Angular.

At philips I had the opportunity to work internationally, regularly visiting our offices in Eindhoven and Aachen. This improved my communication and interpersonal skills.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
